### ISSUE TITLE
Unable to connect to S3 protocol of Dell EMC ECS EX300 

### ISSUE THREAD

**Issue by jax79sg** [triton-inference-server/server#2835]
**Description**
When connecting to Dell EMC EC S EX300 configured with S3 buckets, the error comes out after about 30 seconds
```
error: creating server: Internal - Could not get MetaData for bucket with name 123.123.123.123 due to exception: error message: Unable to connect to endpoint
```
i was able to connect using aws-cli using the exact same connection string. `s3://123.123.123.123/bucketname`

**Triton Information**
What version of Triton are you using?
21.03 docker image, server version 2.8.

Are you using the Triton container or did you build it yourself?
Triton container

**To Reproduce**
Steps to reproduce the behavior.
```
docker run --rm -p 8001:8001 -p 8002:8002 --env envfileContainAccessKeyAndPassword.env nvidia/tritonserver:21.03-py3 tritonserver --model-repository=s3://123.123.123.123:443/bucketname --log-verbose 9
```

Describe the models (framework, inputs, outputs), ideally include the model configuration file (if using an ensemble include the model configuration file for that as well).

Not applicable.

**Expected behavior**

Server running and models loaded

**Comment by CoderHam** [comment]
Thanks for this ticket. We will look into root causing the same.

**Comment by CoderHam** [comment]
@jax79sg can you share the s3 model pat you used? Triton requires the user to specify both host name and port as
`s3://<host_name>:<host_port>/bucketname`. Please refer to the docs here. 
Looks like you used `s3://<host_name>/bucketname` (note the missing port) instead which caused the regex to fail.

**Comment by CoderHam** [comment]
Looks like the path you used was s3://123.123.123.123:443/bucketname which does follow the Triton convention. Was the S3 endpoint http or https?

**Comment by CoderHam** [comment]
@jax79sg is the S3 endpoint HTTPS? I am working on another S3 issue and it appears this might be related.

**Comment by CoderHam** [comment]
We have a couple fixes (#2879 and #2880) that should solve your issue. Feel free to try a build out these patches and get back to us with your feedback. 
Closing this ticket. Please let us know if you still face the problem.

**Comment by jax79sg** [comment]
> We have a couple fixes (#2879 and #2880) that should solve your issue. Feel free to try a build out these patches and get back to us with your feedback.
> Closing this ticket. Please let us know if you still face the problem.

Thanks @CoderHam . Do we need to rebuild from source or we can pull from latest nvcr.io image?

**Comment by CoderHam** [comment]
You will have to rebuild from source. The NGC image for 21.06 should have this fix.
